### ISSUE TITLE
strange behavior - results in wrong inline numbering 

### ISSUE THREAD

**Issue by epogrebnyak** [shyamd/mkdocs-bibtex#107]
I have paper.bib file like:

```
@book{PM18,
  author = {Pearl, Judea and Mackenzie, Dana},
  publisher = {Basic Books},
  title = {The Book of Why: The New Science of Cause and Effect},
  year = {2018},
  address = {New York}
}

@book{Hamilton,
  author = {Hamilton, James D.},
  publisher = {Princeton University Press},
  title = {Time Series Analysis},
  year = {1994},
}
```

and index.md as

```
# Citations

- Book of Why[@PM18]
- Time Series Analysis[@Hamilton]

\bibliography 
```

but the resulting html is not quite right:

```
    Book of Why[1](file:///Q:/bib/site/index.html#fn:1)
    Time Series Analysis[^2]

    James D. Hamilton. Time Series Analysis. Princeton University Press, 1994. [↩](file:///Q:/bib/site/index.html#fnref:1)
```

What can be going wrong? I tried runnng mkdocs on bith Windows and WSL. Many thanks for clues.

**Comment by epogrebnyak** [comment]
In quads third element is always 1:

```
quads = [('[@PM18]', 'PM18', '1', 'Judea Pearl and Dana Mackenzie. *The Book of Why: The New Science of Cause and Effect*. Basic Books, New York, 2018.'), ('[@Hamilton]', 'Hamilton', '1', 'James D. Hamilton. *Time Series Analysis*. Princeton University Press, 1994.')]
```

because of:

https://github.com/shyamd/mkdocs-bibtex/blob/8f73346fae6fb7a41b464505adeb3e671af1b9dd/src/mkdocs_bibtex/plugin.py#L144

so then we get:

```
>>> format_bibliography(quads)
'[^1]: James D. Hamilton. *Time Series Analysis*. Princeton University Press, 1994.'
```

**Comment by epogrebnyak** [comment]
So then here new_bib is always a dictionary like `{"1": ...}`:

https://github.com/shyamd/mkdocs-bibtex/blob/8f73346fae6fb7a41b464505adeb3e671af1b9dd/src/mkdocs_bibtex/utils.py#L169

**Comment by epogrebnyak** [comment]
Here is a new test that fails:

https://github.com/epogrebnyak/mkdocs-bibtex/blob/master/test_files/test_two_items.py

**Comment by shyamd** [comment]
Should be fixed as of v2.0.1 from changes in #108
